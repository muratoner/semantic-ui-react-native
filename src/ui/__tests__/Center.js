import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import React from 'react'
import {StyleSheet} from 'react-native'
import {Center} from '../../'

describe('Center Component', () => {
	it('should render without issues', () => {
		const component = shallow(<Center />)

		expect(component.length).toBe(1)
		const style = StyleSheet.flatten(component.get(0).props.style)
		expect(style).toHaveProperty('alignItems', 'center')
		expect(style).toHaveProperty('justifyContent', 'center')
		expect(toJson(component)).toMatchSnapshot()
	})

	it('should render with flex direction', () => {
		const component = shallow(<Center flexDirection='row' />)

		expect(component.length).toBe(1)
		expect(StyleSheet.flatten(component.get(0).props.style)).toHaveProperty(
			'flexDirection',
			'row'
		)
		expect(toJson(component)).toMatchSnapshot()
	})
})
