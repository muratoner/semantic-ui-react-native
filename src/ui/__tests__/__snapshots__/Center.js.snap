// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Center Component should render with flex direction 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
/>
`;

exports[`Center Component should render without issues 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": undefined,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
/>
`;
